feat: add createdAt and updatedAt fields to EventType model

anikdhabal opened the pull request:
## What does this PR do?

Adds createdAt and updatedAt timestamp fields to the EventType model. The Prisma schema and a migration add the two columns. Adds a Prisma extension (packages/prisma/extensions/event-type-timestamps.ts) to set createdAt on create/createMany.

devin-ai-integration[bot] committed:
feat: add createdAt and updatedAt fields to EventType model

- Add timestamp fields to EventType schema with backward compatibility
- Update all EventType select statements across API v1, v2, and tRPC
- Update platform API output types to include timestamp fields
- Generate migration for existing database records
- Fix test files to handle new timestamp fields properly

The timestamp fields are added as optional (DateTime?) to ensure backward
compatibility - existing EventType records will have null values for these
fields, while new records will automatically get timestamps.

Co-Authored-By: anik@cal.com <adhabal2002@gmail.com>

anikdhabal committed:
Merge branch 'main' into devin/add-event-type-timestamps-1758288266

anikdhabal committed:
Merge branch 'main' into devin/add-event-type-timestamps-1758288266

keithwillcode commented on packages/prisma/migrations/20250919132724_add_event_type_timestamps/migration.sql:
```diff
@@ -0,0 +1,3 @@
+-- AlterTable
+ALTER TABLE "EventType" ADD COLUMN     "createdAt" TIMESTAMP(3) DEFAULT CURRENT_TIMESTAMP,
```
Although - looking more closely, I see you didn't set NOT NULL. Let's confirm behavior.

anikdhabal replied:
done, made changes

devin-ai-integration[bot] committed:
fix: remove @default(now()) from EventType createdAt to prevent existing records from getting current timestamp

- Remove @default(now()) from createdAt field in schema to ensure backward compatibility
- Create migration that only drops DEFAULT constraint without updating existing records
- Existing EventType records will keep null timestamps
- New records will get timestamps via application logic

Co-Authored-By: anik@cal.com <adhabal2002@gmail.com>

devin-ai-integration[bot] committed:
feat: add createdAt and updatedAt timestamps to all EventType creation locations

- Update all seed scripts to set timestamps for new EventType records
- Ensure consistency across all EventType creation patterns
- Maintain backward compatibility with nullable timestamp fields

Co-Authored-By: anik@cal.com <adhabal2002@gmail.com>

devin-ai-integration[bot] committed:
feat: add timestamps to remaining EventType creation locations in test files

- Update integration test files to include createdAt and updatedAt
- Ensure consistency across all EventType creation patterns
- Complete comprehensive update of all EventType creation locations

Co-Authored-By: anik@cal.com <adhabal2002@gmail.com>

anikdhabal committed:
add migration

anikdhabal committed:
few left

Udit-takkar reviewed (dismissed):
@anikdhabal you are passing createdAt, updateAt a lot. We can just add add default(now()) property for createdAt and updatedAt would get automatically updated because of `@updatedAt` property

anikdhabal commented on packages/prisma/schema.prisma:
```diff
@@ -244,6 +244,9 @@ model EventType {
 
   bookingRequiresAuthentication Boolean @default(false)
 
+  createdAt DateTime?
+  updatedAt DateTime? @updatedAt
```
@Udit-takkar In PostgreSQL, when we add a new column to an existing table with a DEFAULT value, all the existing rows will be updated with that default value. To prevent this, wecan add the column without a DEFAULT value and then update it in smaller batches. 
Initially i did that, but reverted it after tested: - https://github.com/calcom/cal.com/pull/23949/commits/8254e1a354e7e9ba489ed6a1639f490e645a1183

anikdhabal committed:
Merge branch 'main' into devin/add-event-type-timestamps-1758288266

Udit-takkar committed:
Merge branch 'main' into devin/add-event-type-timestamps-1758288266

devin-ai-integration[bot] committed:
feat: implement Prisma extension for automatic EventType timestamps

- Create eventTypeTimestampsExtension to automatically set createdAt and updatedAt
- Handle both eventType.create and eventType.createMany operations
- Revert all manual timestamp setting from application code
- Extension only sets timestamps if not already provided
- Maintains backward compatibility - existing records keep null timestamps
- New records get automatic timestamps via Prisma extension
- Follows existing Cal.com extension patterns and architecture

Co-Authored-By: anik@cal.com <adhabal2002@gmail.com>

devin-ai-integration[bot] committed:
feat: remove hardcoded timestamp settings from test files

- Remove manual createdAt/updatedAt settings from routing forms controller test
- Remove manual timestamp settings from event types repository fixture
- Prisma extension now handles timestamps automatically for all EventType operations

Co-Authored-By: anik@cal.com <adhabal2002@gmail.com>

anikdhabal committed:
update

anikdhabal committed:
Update event-type.output.ts

anikdhabal committed:
Merge branch 'main' into devin/add-event-type-timestamps-1758288266

anikdhabal committed:
fix test

anikdhabal committed:
Merge branch 'devin/add-event-type-timestamps-1758288266' of https://github.com/calcom/cal.com into devin/add-event-type-timestamps-1758288266

anikdhabal committed:
fix api v2 build error

anikdhabal committed:
Merge branch 'devin/add-event-type-timestamps-1758288266' of https://github.com/calcom/cal.com into devin/add-event-type-timestamps-1758288266

anikdhabal committed:
update

anikdhabal committed:
revert test

anikdhabal committed:
Merge branch 'main' into devin/add-event-type-timestamps-1758288266

anikdhabal committed:
fix

anikdhabal committed:
Merge branch 'devin/add-event-type-timestamps-1758288266' of https://github.com/calcom/cal.com into devin/add-event-type-timestamps-1758288266

emrysal reviewed (approved):
Good for now 🧑‍🍳  - next PR to remove the event-type-timestamps PR
